### DISCUSSION TITLE
Poor performance of version 7.1 compared to 3.4.4

### DISCUSSION THREAD

**Discussion by meisamhasani** [dotnet/orleans#8457] [General]
Hello
I migrated from version 3.4 to version 7.1 and serialization and configuration went well.
But in production, the performance of the system is not as good as 3.4, and management grain has latency, and after 6 hours, the silos are restarted.
The number of grains is more than 100 thousand

**Comment by ReubenBond** [comment]
Interesting, could you provide more details about your workload so that we can try to determine what you are seeing? Please also submit an issue, since a GitHub discussion is not a great place to work through this kind of thing.

**Comment by meisamhasani** [comment]
Thank you for your answer
I am sending a sample of the error, but maybe it is caused by something else


System.InvalidOperationException: Attempt to access an invalid activation: [Activation: S10.42.3.64:30003:44578723/symbol/F9A71214BEF5F9@205c76e528a04dce9de7515597aa2b0b#Placement=RandomPlacement State=Invalid]
         at Orleans.Runtime.GrainRuntime.<CheckRuntimeContext>g__ThrowInvalidActivation|20_1(ActivationData activationData) in /_/src/Orleans.Runtime/Core/GrainRuntime.cs:line 101
         at Orleans.Runtime.GrainRuntime.CheckRuntimeContext(IGrainContext context) in /_/src/Orleans.Runtime/Core/GrainRuntime.cs:line 100
         at Orleans.Core.StateStorageBridge`1.get_State() in /_/src/Orleans.Runtime/Storage/StateStorageBridge.cs:line 30
'System.InvalidOperationException:' is not recognized as an internal or external command,
operable program or batch file.

**Comment by meisamhasani** [comment]
I'm looking for it to be clear if I made a mistake in the 7.1 config because I don't want to stay on the 3.4 version forever.

**Comment by meisamhasani** [comment]
i used:
UseAdoNetClustering & AddMemoryGrainStorage
I did not use :
UseAdoNetReminderService or AddAdoNetGrainStorage

**Comment by meisamhasani** [comment]
![image](https://github.com/dotnet/orleans/assets/2770416/dad87ade-94e1-453c-881d-cfeaf18cdfc1)
![image](https://github.com/dotnet/orleans/assets/2770416/0ce1ef61-5174-41f1-86c4-8b5fe1236666)
![image](https://github.com/dotnet/orleans/assets/2770416/30955ff2-9b49-44db-b30e-44baeaac9d27)

**Comment by ReubenBond** [comment]
@meisamhasani I have converted this into an issue for you: #8458. Discussions are not good for discussion (they are for Stack Overflow style Q&A)

**Reply by meisamhasani** [reply]
thank you . Excellent
